internal/deploy/validations.go

```diff
@@ -0,0 +1,155 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+	kerrs "github.com/shipa-corp/ketch/internal/errors"
+	"github.com/shipa-corp/ketch/internal/validation"
+)
+
+type statusType int
+
+const (
+	missingValue statusType = iota
+	invalidValue
+)
+
+type statusError struct {
```

Comment:
It seems like it might be worthwhile to share this error formatting with other packages. What do you think?